### PR TITLE
[Security][TokenInterface] Prepare for the new serialization mechanism

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Continuation of https://github.com/symfony/symfony/pull/30965

### CODE REVIEW

File: src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php [hunk 1]
```diff
@@ -19,7 +19,9 @@
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Johannes M. Schmitt <schmittjoh@gmail.com>
  *
- * @method string[] getRoleNames() The associated roles - not implementing it is deprecated since Symfony 4.3
+ * @method array __serialize Returns all the necessary state of the object for serialization purposes - not implementing it is deprecated since Symfony 4.3 and will be required in Symfony 5.0.
+ * @method void     __unserialize(array $data) Restores the object state from an array given by __serialize() - not implementing it is deprecated since Symfony 4.3 and will be required in Symfony 5.0.
+ * @method string[] getRoleNames()                                                                            The associated roles - not implementing it is deprecated since Symfony 4.3
```
Comment (fabpot) [inline src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php]: Indentation looks weird.

File: UPGRADE-4.3.md [hunk 2]
```diff
@@ -153,6 +153,8 @@ Security
 
  * Using `Argon2iPasswordEncoder` while only the `argon2id` algorithm is supported
    is deprecated, use `Argon2idPasswordEncoder` instead
+ * Not implementing the methods `__serialize` and `_unserialize` in classes implementing
```
Comment (nicolas-grekas) [inline UPGRADE-4.3.md]: missing `_` before `_unserialize`

File: src/Symfony/Component/Security/CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +22,8 @@ CHANGELOG
  * Added `Argon2idPasswordEncoder`
  * Deprecated using `Argon2iPasswordEncoder` while only the `argon2id` algorithm
    is supported, use `Argon2idPasswordEncoder` instead
+ * Added two virtual methods (`__serialize` and `__unserialize`) to the `TokenInterface`
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/CHANGELOG.md]: Added methods `__serialize` and `__unserialize` to the `TokenInterface` ("virtual" is not needed imho - we describe the intent here, technically we follow our BC promise of course)

**Review by nicolas-grekas — APPROVED** [review]
once comments are addressed

File: src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php (hunk 1 above)
Comment (fancyweb) [reply src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php]: Indentation was changed by the fabbot patch.

Comment (fancyweb) [reply src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php]: But yeah actually it is really weird 😕

Comment (nicolas-grekas) [reply src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php]: please make it the nest you can fabbot is just a tool, it doesn't have the last word

Comment (chalasr) [reply src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php]: fixed

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @fancyweb.

**Pull request merged**